### PR TITLE
enable bfloat16 for cpu

### PR DESCRIPTION
**Pull request by sywangyi**
if there's no cuda. disable custom kernels

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sywangyi** [comment]
@jianan-gu

**Comment by sywangyi** [comment]
@Narsil could you please help review the PR? I run the TGI in cpu only platform, and fix some issues

File: server/text_generation_server/models/bloom.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __init__(
             dtype = torch.float16 if dtype is None else dtype
         else:
             device = torch.device("cpu")
-            dtype = torch.float32
+            dtype = torch.bfloat16 if dtype == torch.bfloat16 else torch.float32
```
Comment (Narsil) [inline server/text_generation_server/models/bloom.py:54]:
```suggestion
            dtype = torch.float32 if dtype is None else dtype
```

Can you modify everywhere accordingly ?

Comment (sywangyi) [reply server/text_generation_server/models/bloom.py:54]: cpu doesnot support float16

Comment (sywangyi) [reply server/text_generation_server/models/bloom.py]: done

**Review by Narsil — APPROVED** [review]
LGTM

**Pull request merged**